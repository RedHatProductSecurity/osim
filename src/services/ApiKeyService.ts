//    <div class="alert alert-danger"
//          role="alert"
//         v-if="!/.+/.test(settingsStore.settings.bugzillaApiKey ?? '')">
//       You have not set your Bugzilla API key in this tab!<br/>
//       Flaw creation requires your Bugzilla API key to be set.<br/>
//       Visit <RouterLink :to="{name: 'settings'}">Settings</RouterLink>.
//     </div>

import { useSettingsStore } from '@/stores/SettingsStore';
import { useToastStore } from '@/stores/ToastStore';
// import { useRouter } from 'vue-router';

export function notifyApiKeyUnset() {

  const { settings: { bugzillaApiKey, jiraApiKey } } = useSettingsStore();
  const unsetKeys: string[] = [];

  if (!bugzillaApiKey) {
    unsetKeys.push('Bugzilla');
  }

  if (!jiraApiKey) {
    unsetKeys.push('JIRA');
  }

  if (!bugzillaApiKey || !jiraApiKey) {
    const { addToast } = useToastStore();
    const lis = unsetKeys.map((key) => `<li>${key}</li>`).join('');

    addToast({
      css: 'warning',
      title: 'Read-Only Mode',
      bodyHtml: true,
      body: `You have not set the following keys in this tab: <ul>${lis}</ul>` +
      'Flaw creation requires your Bugzilla API or JIRA key to be set. ' +
      'Visit Settings to set any required keys.'
    });
    // maybeRedirectToSettings();
  }
}

// function maybeRedirectToSettings () {
//   const router = useRouter();
//   if (router.currentRoute.value.fullPath === '/flaws/new') {
//     router.push({path: '/settings'});
//   }
// }

