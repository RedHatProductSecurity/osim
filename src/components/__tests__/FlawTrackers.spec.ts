import { describe, expect } from 'vitest';

import FlawTrackers from '@/components/FlawTrackers.vue';
import sampleTrackersQueryResult from '@/components/__tests__/__fixtures__/sampleTrackersQueryResult.json';

import { getTrackersForFlaws } from '@/services/TrackerService';
import type { ZodFlawType } from '@/types';
import { mountWithConfig } from '@/__tests__/helpers';

import { osimFullFlawTest, osimRequiredFlawTest } from './test-suite-helpers';

// vi.mock('@/composables/useTrackers', () => ({
//   useTrackers: vi.fn().mockReturnValue({
//     trackersToFile: [],
//     isLoadingTrackers: false,
//   }),
// }));

const mountFlawTrackers = (props: InstanceType<typeof FlawTrackers>['$props']) => mountWithConfig(FlawTrackers, {
  props,
});
// createTestingPinia();

vi.mock('@/services/TrackerService');

// const globalOptions = {
//   global: {
//     stubs: {
//       RouterLink: true,
//     },
//     directives: {
//       'osim-loading': LoadingAnimationDirective,
//     },
//   },
// };

describe('flawTrackers', () => {
  beforeEach(() => {
    vi.mocked(getTrackersForFlaws).mockResolvedValue(sampleTrackersQueryResult);
  });
  osimRequiredFlawTest('Correctly renders the component when there are not trackers to display', async ({ flaw }) => {
    const subject = mountFlawTrackers({
      // ...globalOptions,
      // props: {
      flaw: flaw as ZodFlawType,
      relatedFlaws: [flaw as ZodFlawType],
      displayedTrackers: [],
      allTrackersCount: 0,
      // },
    });
    expect(subject.html()).toMatchSnapshot();
  });

  osimFullFlawTest('Correctly renders the component when there are trackers to display', async ({ flaw }) => {
    const subject = mountFlawTrackers({
      // ...globalOptions,
      // props: {
      flaw: flaw as ZodFlawType,
      relatedFlaws: [flaw as ZodFlawType],
      displayedTrackers: flaw.affects
        .flatMap(affect => affect.trackers
          .map(tracker => ({ ...tracker, ps_module: affect.ps_module })),
        ),
      allTrackersCount: 0,
      // },
    });
    expect(subject.html()).toMatchSnapshot();
  });

  osimFullFlawTest('Per page setting correctly changes the table page items number', async ({ flaw }) => {
    const subject = mountFlawTrackers({
      // ...globalOptions,
      // props: {
      flaw: flaw as ZodFlawType,
      relatedFlaws: [flaw as ZodFlawType],
      displayedTrackers: flaw.affects
        .flatMap(affect => affect.trackers
          .map(tracker => ({ ...tracker, ps_module: affect.ps_module })),
        ),
      allTrackersCount: 0,
      // },
    });

    let trackersTableRows = subject.findAll('.affects-trackers .osim-tracker-card table tbody tr');
    let rowCount = trackersTableRows.length;
    expect(rowCount).toBe(6);
    let itemsPerPageIndicator = subject.find('.trackers-toolbar .tracker-badges .btn span');
    expect(itemsPerPageIndicator.text()).toBe('Per page: 10');

    const reduceItemsBtn = subject.find('.trackers-toolbar .tracker-badges .btn .bi-dash-square');
    for (let i = 0; i < 5; i++) {
      await reduceItemsBtn.trigger('click');
    }

    trackersTableRows = subject.findAll('.affects-trackers .osim-tracker-card table tbody tr');
    rowCount = trackersTableRows.length;
    expect(rowCount).toBe(5);
    itemsPerPageIndicator = subject.find('.trackers-toolbar .tracker-badges .btn span');
    expect(itemsPerPageIndicator.text()).toBe('Per page: 5');

    const increaseItemsBtn = subject.find('.trackers-toolbar .tracker-badges .btn .bi-plus-square');
    for (let i = 0; i < 5; i++) {
      await increaseItemsBtn.trigger('click');
    }

    trackersTableRows = subject.findAll('.affects-trackers .osim-tracker-card table tbody tr');
    rowCount = trackersTableRows.length;
    expect(rowCount).toBe(6);
    itemsPerPageIndicator = subject.find('.trackers-toolbar .tracker-badges .btn span');
    expect(itemsPerPageIndicator.text()).toBe('Per page: 10');
  });

  osimFullFlawTest('Trackers can be sorted by clicking on the date field columns', async ({ flaw }) => {
    const subject = mountFlawTrackers({
      // ...globalOptions,
      // props: {
      flaw: flaw as ZodFlawType,
      relatedFlaws: [flaw as ZodFlawType],
      displayedTrackers: flaw.affects
        .flatMap(affect => affect.trackers
          .map(tracker => ({ ...tracker, ps_module: affect.ps_module })),
        ),
      allTrackersCount: 0,
      // },
    });

    let trackersTableRows = subject.findAll('.affects-trackers .osim-tracker-card table tbody tr');
    let firstTracker = trackersTableRows[0];
    expect(firstTracker.find('td:nth-of-type(3)').text()).toBe('xxxx-0-006');

    const componentHeader = subject.find('.affects-trackers table thead tr th:nth-of-type(6)');
    expect(componentHeader.exists()).toBe(true);
    expect(componentHeader.text()).toBe('Created date');
    await componentHeader.trigger('click');

    trackersTableRows = subject.findAll('.affects-trackers .osim-tracker-card table tbody tr');
    firstTracker = trackersTableRows[0];
    expect(firstTracker.find('td:nth-of-type(3)').text()).toBe('xxxx-0-001');
  });

  osimFullFlawTest('Trackers display functional external links', async ({ flaw }) => {
    const subject = mountFlawTrackers({
      flaw,
      relatedFlaws: [flaw as ZodFlawType],
      displayedTrackers: flaw.affects
        .flatMap(affect => affect.trackers
          .map(tracker => ({ ...tracker, ps_module: affect.ps_module })),
        ),
      allTrackersCount: 0,
    });

    const trackersTableRows = subject.findAll('.affects-trackers .osim-tracker-card table tbody tr');
    const firstTracker = trackersTableRows[0];

    const trackerLink = firstTracker.find('td:nth-child(1) > a');
    expect(trackerLink.attributes('href')).toBe('http://jira-service:8002/browse/XXXX-0006');
  });

  osimFullFlawTest('Tracker modules table cell have correct tooltip', async ({ flaw }) => {
    const subject = mountFlawTrackers({
      flaw,
      relatedFlaws: [flaw as ZodFlawType],
      displayedTrackers: flaw.affects
        .flatMap(affect => affect.trackers
          .map(tracker => ({ ...tracker, ps_module: affect.ps_module })),
        ),
      allTrackersCount: 0,
    });
    const trackerRow = subject.findAll('.affects-trackers .osim-tracker-card table tbody tr')[1];
    const trackerModuleCell = trackerRow.find('td:nth-of-type(2)');
    expect(trackerModuleCell.attributes('title')).toBe('openshift-5');
  });

  osimFullFlawTest('Tracker ps_stream table cell have correct tooltip', async ({ flaw }) => {
    const subject = mountFlawTrackers({
      flaw,
      relatedFlaws: [flaw as ZodFlawType],
      displayedTrackers: flaw.affects
        .flatMap(affect => affect.trackers
          .map(tracker => ({ ...tracker, ps_module: affect.ps_module })),
        ),
      allTrackersCount: 0,
    });
    const trackerRow = subject.findAll('.affects-trackers .osim-tracker-card table tbody tr')[0];
    const trackerModuleCell = trackerRow.find('td:nth-of-type(3)');
    expect(trackerModuleCell.attributes('title')).toBe('xxxx-0-006');
  });

  osimFullFlawTest('Displays embedded trackers manager', async ({ flaw }) => {
    const subject = mountFlawTrackers({
      // ...globalOptions,
      // props: {
      flaw: flaw as ZodFlawType,
      relatedFlaws: [flaw as ZodFlawType],
      displayedTrackers: flaw.affects
        .flatMap(affect => affect.trackers
          .map(tracker => ({ ...tracker, ps_module: affect.ps_module })),
        ),
      allTrackersCount: 0,
      // },
    });

    let trackerManagerElement = subject.find('.osim-tracker-manager');
    expect(trackerManagerElement.exists()).toBe(false);

    let toggleTrackerManagerView = subject.find('.trackers-toolbar .btn-info');
    expect(toggleTrackerManagerView.text()).toBe('Show Trackers Manager');
    await toggleTrackerManagerView.trigger('click');

    trackerManagerElement = subject.find('.osim-tracker-manager');
    expect(trackerManagerElement.exists()).toBe(true);

    toggleTrackerManagerView = subject.find('.trackers-toolbar .btn-info');
    expect(toggleTrackerManagerView.text()).toBe('Hide Trackers Manager');

    expect(subject.html()).toMatchSnapshot();
  });
});
