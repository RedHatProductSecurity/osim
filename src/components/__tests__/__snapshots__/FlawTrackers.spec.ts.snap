// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`flawTrackers > Correctly renders the component when there are not trackers to display 1`] = `
"<div data-v-7006d35c="">
  <div data-v-7006d35c="" class="affects-trackers">
    <div data-v-7006d35c="" class="bg-dark rounded-top-2 text-info">
      <div data-v-7006d35c="" class="affect-trackers-heading d-flex p-1 pt-2 px-3">
        <h5 data-v-7006d35c="" class="m-0">Trackers</h5>
      </div>
      <!--v-if-->
      <div data-v-7006d35c="" class="trackers-toolbar p-2 pt-1">
        <div data-v-7006d35c="" class="tracker-badges">
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div><button data-v-7006d35c="" type="button" class="btn btn-sm btn-info ms-auto"><i data-v-7006d35c="" class="bi bi-binoculars"></i> Show Trackers Manager </button>
      </div>
    </div>
    <div data-v-7006d35c="" class="osim-tracker-card pb-2 pt-0 pe-2 ps-2 bg-dark">
      <table data-v-7006d35c="" class="table table-striped table-info mb-0">
        <thead data-v-7006d35c="" class="sticky-top">
          <tr data-v-7006d35c="">
            <th data-v-7006d35c="">Bug ID</th>
            <th data-v-7006d35c="">Module</th>
            <th data-v-7006d35c="">Product Stream</th>
            <th data-v-7006d35c=""><span data-v-7006d35c="" class="align-bottom me-1">Status</span><button data-v-7006d35c="" id="status-filter" type="button" class="btn btn-sm border-0 p-0" data-bs-toggle="dropdown" data-bs-auto-close="outside" aria-expanded="false"><i data-v-7006d35c="" class="bi bi-funnel" title=""></i></button>
              <ul data-v-7006d35c="" class="dropdown-menu" aria-labelledby="status-filter"></ul>
            </th>
            <th data-v-7006d35c="">Resolution</th>
            <th data-v-7006d35c=""> Created date <i data-v-7006d35c="" class="bi-caret-up-fill bi"></i></th>
            <th data-v-7006d35c=""> Updated date <i data-v-7006d35c="" class="opacity-0 bi-caret-up-fill bi"></i></th>
          </tr>
        </thead>
        <tbody data-v-7006d35c=""></tbody>
      </table>
      <div data-v-7006d35c=""> No trackers to display </div>
    </div>
  </div>
  <!--v-if-->
</div>"
`;

exports[`flawTrackers > Correctly renders the component when there are trackers to display 1`] = `
"<div data-v-7006d35c="">
  <div data-v-7006d35c="" class="affects-trackers">
    <div data-v-7006d35c="" class="bg-dark rounded-top-2 text-info">
      <div data-v-7006d35c="" class="affect-trackers-heading d-flex p-1 pt-2 px-3">
        <h5 data-v-7006d35c="" class="m-0">Trackers</h5>
      </div>
      <!--v-if-->
      <div data-v-7006d35c="" class="trackers-toolbar p-2 pt-1">
        <div data-v-7006d35c="" class="tracker-badges">
          <div data-v-7006d35c="" class="per-page-btn btn btn-sm badge"><i data-v-7006d35c="" style="pointer-events: auto;" class="bi bi-dash-square fs-6 my-auto" title="Reduce trackers per page"></i><span data-v-7006d35c="" class="mx-2 my-auto">Per page: 10</span><i data-v-7006d35c="" style="pointer-events: auto;" class="bi bi-plus-square fs-6 my-auto" title="Increase trackers per page"></i></div>
          <!--v-if-->
          <div data-v-7006d35c="" class="badge" title="Trackers displayed in current page"><span data-v-7006d35c="">Displaying 6</span></div>
        </div><button data-v-7006d35c="" type="button" class="btn btn-sm btn-info ms-auto"><i data-v-7006d35c="" class="bi bi-binoculars"></i> Show Trackers Manager </button>
      </div>
    </div>
    <div data-v-7006d35c="" class="osim-tracker-card pb-2 pt-0 pe-2 ps-2 bg-dark">
      <table data-v-7006d35c="" class="table table-striped table-info mb-0">
        <thead data-v-7006d35c="" class="sticky-top">
          <tr data-v-7006d35c="">
            <th data-v-7006d35c="">Bug ID</th>
            <th data-v-7006d35c="">Module</th>
            <th data-v-7006d35c="">Product Stream</th>
            <th data-v-7006d35c=""><span data-v-7006d35c="" class="align-bottom me-1">Status</span><button data-v-7006d35c="" id="status-filter" type="button" class="btn btn-sm border-0 p-0" data-bs-toggle="dropdown" data-bs-auto-close="outside" aria-expanded="false"><i data-v-7006d35c="" class="bi bi-funnel" title=""></i></button>
              <ul data-v-7006d35c="" class="dropdown-menu" aria-labelledby="status-filter">
                <li data-v-7006d35c=""><a data-v-7006d35c="" href="#" class="btn py-0 dropdown-item"><input data-v-7006d35c="" type="checkbox" class="form-check-input me-2"><span data-v-7006d35c="">NEW</span></a></li>
                <li data-v-7006d35c=""><a data-v-7006d35c="" href="#" class="btn py-0 dropdown-item"><input data-v-7006d35c="" type="checkbox" class="form-check-input me-2"><span data-v-7006d35c="">CLOSED</span></a></li>
              </ul>
            </th>
            <th data-v-7006d35c="">Resolution</th>
            <th data-v-7006d35c=""> Created date <i data-v-7006d35c="" class="bi-caret-up-fill bi"></i></th>
            <th data-v-7006d35c=""> Updated date <i data-v-7006d35c="" class="opacity-0 bi-caret-up-fill bi"></i></th>
          </tr>
        </thead>
        <tbody data-v-7006d35c="">
          <tr data-v-7006d35c="">
            <td data-v-7006d35c=""><a data-v-7006d35c="" target="_blank">XXXX-0006 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="" title="openshift-5">openshift-5</td>
            <td data-v-7006d35c="" title="xxxx-0-006">xxxx-0-006</td>
            <td data-v-7006d35c=""><a data-v-7006d35c="" href="http://jira-service:8002/browse/XXXX-0006" target="_blank">XXXX-0006 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="">openshift-5</td>
            <td data-v-7006d35c="">xxxx-0-006</td>
            <td data-v-7006d35c="">CLOSED</td>
            <td data-v-7006d35c=""></td>
            <td data-v-7006d35c="">2024-07-01 13:05 UTC</td>
            <td data-v-7006d35c="">2024-07-01 13:06 UTC</td>
          </tr>
          <tr data-v-7006d35c="">
            <td data-v-7006d35c=""><a data-v-7006d35c="" target="_blank">XXXX-0005 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="" title="openshift-5">openshift-5</td>
            <td data-v-7006d35c="" title="xxxx-0-005">xxxx-0-005</td>
            <td data-v-7006d35c=""><a data-v-7006d35c="" href="http://jira-service:8002/browse/XXXX-0005" target="_blank">XXXX-0005 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="">openshift-5</td>
            <td data-v-7006d35c="">xxxx-0-005</td>
            <td data-v-7006d35c="">NEW</td>
            <td data-v-7006d35c=""></td>
            <td data-v-7006d35c="">2024-08-01 13:05 UTC</td>
            <td data-v-7006d35c="">2024-08-01 13:06 UTC</td>
          </tr>
          <tr data-v-7006d35c="">
            <td data-v-7006d35c=""><a data-v-7006d35c="" target="_blank">XXXX-0004 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="" title="openshift-1">openshift-1</td>
            <td data-v-7006d35c="" title="xxxx-0-004">xxxx-0-004</td>
            <td data-v-7006d35c=""><a data-v-7006d35c="" href="http://jira-service:8002/browse/XXXX-0004" target="_blank">XXXX-0004 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="">openshift-1</td>
            <td data-v-7006d35c="">xxxx-0-004</td>
            <td data-v-7006d35c="">CLOSED</td>
            <td data-v-7006d35c=""></td>
            <td data-v-7006d35c="">2024-09-01 13:05 UTC</td>
            <td data-v-7006d35c="">2024-09-01 13:06 UTC</td>
          </tr>
          <tr data-v-7006d35c="">
            <td data-v-7006d35c=""><a data-v-7006d35c="" target="_blank">XXXX-0003 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="" title="openshift-1">openshift-1</td>
            <td data-v-7006d35c="" title="xxxx-0-003">xxxx-0-003</td>
            <td data-v-7006d35c=""><a data-v-7006d35c="" href="http://jira-service:8002/browse/XXXX-0003" target="_blank">XXXX-0003 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="">openshift-1</td>
            <td data-v-7006d35c="">xxxx-0-003</td>
            <td data-v-7006d35c="">NEW</td>
            <td data-v-7006d35c=""></td>
            <td data-v-7006d35c="">2024-10-01 13:05 UTC</td>
            <td data-v-7006d35c="">2024-10-01 13:06 UTC</td>
          </tr>
          <tr data-v-7006d35c="">
            <td data-v-7006d35c=""><a data-v-7006d35c="" target="_blank">XXXX-0002 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="" title="openshift-1">openshift-1</td>
            <td data-v-7006d35c="" title="xxxx-0-002">xxxx-0-002</td>
            <td data-v-7006d35c=""><a data-v-7006d35c="" href="http://jira-service:8002/browse/XXXX-0002" target="_blank">XXXX-0002 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="">openshift-1</td>
            <td data-v-7006d35c="">xxxx-0-002</td>
            <td data-v-7006d35c="">CLOSED</td>
            <td data-v-7006d35c=""></td>
            <td data-v-7006d35c="">2024-11-01 13:05 UTC</td>
            <td data-v-7006d35c="">2024-11-01 13:06 UTC</td>
          </tr>
          <tr data-v-7006d35c="">
            <td data-v-7006d35c=""><a data-v-7006d35c="" target="_blank">XXXX-0001 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="" title="openshift-1">openshift-1</td>
            <td data-v-7006d35c="" title="xxxx-0-001">xxxx-0-001</td>
            <td data-v-7006d35c=""><a data-v-7006d35c="" href="http://jira-service:8002/browse/XXXX-0001" target="_blank">XXXX-0001 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="">openshift-1</td>
            <td data-v-7006d35c="">xxxx-0-001</td>
            <td data-v-7006d35c="">NEW</td>
            <td data-v-7006d35c=""></td>
            <td data-v-7006d35c="">2024-12-01 13:05 UTC</td>
            <td data-v-7006d35c="">2024-12-01 13:06 UTC</td>
          </tr>
        </tbody>
      </table>
      <!--v-if-->
    </div>
  </div>
  <!--v-if-->
</div>"
`;

exports[`flawTrackers > Displays embedded trackers manager 1`] = `
"<div data-v-7006d35c="">
  <div data-v-7006d35c="" class="affects-trackers">
    <div data-v-7006d35c="" class="bg-dark rounded-top-2 text-info">
      <div data-v-7006d35c="" class="affect-trackers-heading d-flex p-1 pt-2 px-3">
        <h5 data-v-7006d35c="" class="m-0">Trackers</h5>
      </div>
      <!--v-if-->
      <div data-v-7006d35c="" class="trackers-toolbar p-2 pt-1">
        <div data-v-7006d35c="" class="tracker-badges">
          <div data-v-7006d35c="" class="per-page-btn btn btn-sm badge"><i data-v-7006d35c="" style="pointer-events: auto;" class="bi bi-dash-square fs-6 my-auto" title="Reduce trackers per page"></i><span data-v-7006d35c="" class="mx-2 my-auto">Per page: 10</span><i data-v-7006d35c="" style="pointer-events: auto;" class="bi bi-plus-square fs-6 my-auto" title="Increase trackers per page"></i></div>
          <!--v-if-->
          <div data-v-7006d35c="" class="badge" title="Trackers displayed in current page"><span data-v-7006d35c="">Displaying 6</span></div>
        </div><button data-v-7006d35c="" type="button" class="btn btn-sm btn-info ms-auto"><i data-v-7006d35c="" class="bi bi-eye-slash-fill"></i> Hide Trackers Manager </button>
      </div>
    </div>
    <div data-v-7006d35c="" class="osim-tracker-card pb-2 pt-0 pe-2 ps-2 bg-dark">
      <table data-v-7006d35c="" class="table table-striped table-info mb-0">
        <thead data-v-7006d35c="" class="sticky-top">
          <tr data-v-7006d35c="">
            <th data-v-7006d35c="">Bug ID</th>
            <th data-v-7006d35c="">Module</th>
            <th data-v-7006d35c="">Product Stream</th>
            <th data-v-7006d35c=""><span data-v-7006d35c="" class="align-bottom me-1">Status</span><button data-v-7006d35c="" id="status-filter" type="button" class="btn btn-sm border-0 p-0" data-bs-toggle="dropdown" data-bs-auto-close="outside" aria-expanded="false"><i data-v-7006d35c="" class="bi bi-funnel" title=""></i></button>
              <ul data-v-7006d35c="" class="dropdown-menu" aria-labelledby="status-filter">
                <li data-v-7006d35c=""><a data-v-7006d35c="" href="#" class="btn py-0 dropdown-item"><input data-v-7006d35c="" type="checkbox" class="form-check-input me-2"><span data-v-7006d35c="">NEW</span></a></li>
                <li data-v-7006d35c=""><a data-v-7006d35c="" href="#" class="btn py-0 dropdown-item"><input data-v-7006d35c="" type="checkbox" class="form-check-input me-2"><span data-v-7006d35c="">CLOSED</span></a></li>
              </ul>
            </th>
            <th data-v-7006d35c="">Resolution</th>
            <th data-v-7006d35c=""> Created date <i data-v-7006d35c="" class="bi-caret-up-fill bi"></i></th>
            <th data-v-7006d35c=""> Updated date <i data-v-7006d35c="" class="opacity-0 bi-caret-up-fill bi"></i></th>
          </tr>
        </thead>
        <tbody data-v-7006d35c="">
          <tr data-v-7006d35c="">
            <td data-v-7006d35c=""><a data-v-7006d35c="" target="_blank">XXXX-0006 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="" title="openshift-5">openshift-5</td>
            <td data-v-7006d35c="" title="xxxx-0-006">xxxx-0-006</td>
            <td data-v-7006d35c=""><a data-v-7006d35c="" href="http://jira-service:8002/browse/XXXX-0006" target="_blank">XXXX-0006 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="">openshift-5</td>
            <td data-v-7006d35c="">xxxx-0-006</td>
            <td data-v-7006d35c="">CLOSED</td>
            <td data-v-7006d35c=""></td>
            <td data-v-7006d35c="">2024-07-01 13:05 UTC</td>
            <td data-v-7006d35c="">2024-07-01 13:06 UTC</td>
          </tr>
          <tr data-v-7006d35c="">
            <td data-v-7006d35c=""><a data-v-7006d35c="" target="_blank">XXXX-0005 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="" title="openshift-5">openshift-5</td>
            <td data-v-7006d35c="" title="xxxx-0-005">xxxx-0-005</td>
            <td data-v-7006d35c=""><a data-v-7006d35c="" href="http://jira-service:8002/browse/XXXX-0005" target="_blank">XXXX-0005 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="">openshift-5</td>
            <td data-v-7006d35c="">xxxx-0-005</td>
            <td data-v-7006d35c="">NEW</td>
            <td data-v-7006d35c=""></td>
            <td data-v-7006d35c="">2024-08-01 13:05 UTC</td>
            <td data-v-7006d35c="">2024-08-01 13:06 UTC</td>
          </tr>
          <tr data-v-7006d35c="">
            <td data-v-7006d35c=""><a data-v-7006d35c="" target="_blank">XXXX-0004 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="" title="openshift-1">openshift-1</td>
            <td data-v-7006d35c="" title="xxxx-0-004">xxxx-0-004</td>
            <td data-v-7006d35c=""><a data-v-7006d35c="" href="http://jira-service:8002/browse/XXXX-0004" target="_blank">XXXX-0004 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="">openshift-1</td>
            <td data-v-7006d35c="">xxxx-0-004</td>
            <td data-v-7006d35c="">CLOSED</td>
            <td data-v-7006d35c=""></td>
            <td data-v-7006d35c="">2024-09-01 13:05 UTC</td>
            <td data-v-7006d35c="">2024-09-01 13:06 UTC</td>
          </tr>
          <tr data-v-7006d35c="">
            <td data-v-7006d35c=""><a data-v-7006d35c="" target="_blank">XXXX-0003 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="" title="openshift-1">openshift-1</td>
            <td data-v-7006d35c="" title="xxxx-0-003">xxxx-0-003</td>
            <td data-v-7006d35c=""><a data-v-7006d35c="" href="http://jira-service:8002/browse/XXXX-0003" target="_blank">XXXX-0003 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="">openshift-1</td>
            <td data-v-7006d35c="">xxxx-0-003</td>
            <td data-v-7006d35c="">NEW</td>
            <td data-v-7006d35c=""></td>
            <td data-v-7006d35c="">2024-10-01 13:05 UTC</td>
            <td data-v-7006d35c="">2024-10-01 13:06 UTC</td>
          </tr>
          <tr data-v-7006d35c="">
            <td data-v-7006d35c=""><a data-v-7006d35c="" target="_blank">XXXX-0002 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="" title="openshift-1">openshift-1</td>
            <td data-v-7006d35c="" title="xxxx-0-002">xxxx-0-002</td>
            <td data-v-7006d35c=""><a data-v-7006d35c="" href="http://jira-service:8002/browse/XXXX-0002" target="_blank">XXXX-0002 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="">openshift-1</td>
            <td data-v-7006d35c="">xxxx-0-002</td>
            <td data-v-7006d35c="">CLOSED</td>
            <td data-v-7006d35c=""></td>
            <td data-v-7006d35c="">2024-11-01 13:05 UTC</td>
            <td data-v-7006d35c="">2024-11-01 13:06 UTC</td>
          </tr>
          <tr data-v-7006d35c="">
            <td data-v-7006d35c=""><a data-v-7006d35c="" target="_blank">XXXX-0001 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="" title="openshift-1">openshift-1</td>
            <td data-v-7006d35c="" title="xxxx-0-001">xxxx-0-001</td>
            <td data-v-7006d35c=""><a data-v-7006d35c="" href="http://jira-service:8002/browse/XXXX-0001" target="_blank">XXXX-0001 <i data-v-7006d35c="" class="bi-box-arrow-up-right"></i></a></td>
            <td data-v-7006d35c="">openshift-1</td>
            <td data-v-7006d35c="">xxxx-0-001</td>
            <td data-v-7006d35c="">NEW</td>
            <td data-v-7006d35c=""></td>
            <td data-v-7006d35c="">2024-12-01 13:05 UTC</td>
            <td data-v-7006d35c="">2024-12-01 13:06 UTC</td>
          </tr>
        </tbody>
      </table>
      <!--v-if-->
    </div>
  </div>
  <div data-v-4977998e="" data-v-7006d35c="" class="trackers-manager py-3 px-4 embedded">
    <h4 data-v-4977998e="" class="mb-2 d-flex"> Trackers Manager </h4>
    <div data-v-4977998e="" class="trackers-search-bar gap-2"><input data-v-4977998e="" type="text" class="form-control form-control-sm" placeholder="Filter by stream or component name"><button data-v-4977998e="" type="button" class="btn btn-sm btn-black"><i data-v-4977998e="" class="bi bi-check-all"></i> Select All </button><button data-v-4977998e="" type="button" class="btn btn-sm btn-black"><i data-v-4977998e="" class="bi bi-x-square"></i> Deselect All </button></div>
    <div class="row mt-3" data-v-4977998e="">
      <div class="col" data-v-4977998e="">
        <h5 class="d-inline-block fs-5" data-v-4977998e="">Available Trackers</h5><span class="fst-italic" data-v-4977998e=""> (<i class="bi bi-box-arrow-in-right" data-v-4977998e=""></i> indicates a Suggested Tracker) </span>
      </div>
      <div class="col" data-v-4977998e="">
        <h6 class="fs-5" data-v-4977998e="">Selected To File</h6>
      </div>
    </div>
    <div data-v-4977998e="" class="row">
      <div data-v-4977998e="" class="col">
        <div data-v-4977998e="" class="mb-2">
          <!--v-if-->
          <div data-v-4977998e="" class="osim-tracker-list"></div>
        </div>
      </div>
      <div data-v-4977998e="" class="col">
        <div data-v-4977998e="" class="mb-2">
          <!--v-if-->
          <div data-v-4977998e="" class="osim-tracker-list"></div>
        </div>
      </div>
    </div>
    <div data-v-4977998e="" class="row mt-4">
      <div data-v-4977998e="" class="col mb-2">
        <h5 data-v-4977998e="">Already Filed</h5>
        <!--v-if-->
        <div data-v-4977998e="" class="osim-tracker-list"></div>
      </div>
      <div data-v-4977998e="" class="col">
        <div data-v-4977998e="">
          <h5 data-v-4977998e="">Untrackable Affects <i data-v-4977998e="" class="bi bi-exclamation-circle text-danger" title="These affects do not have available trackers. This may indicate an issue with product defintions. Please contact the OSIDB/OSIM team for assistance."></i></h5>
          <!--v-if-->
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
    </div><button data-v-4977998e="" type="button" class="btn btn-sm btn-black osim-file-trackers mt-4" disabled=""><i data-v-4977998e="" class="bi bi-archive"></i> File Selected Trackers </button>
  </div>
</div>"
`;
